apps/api/src/app/layouts/layouts.controller.ts

```diff
@@ -1,13 +1,49 @@
-import { Controller, Inject, UseGuards } from '@nestjs/common';
-import { ApiTags } from '@nestjs/swagger';
+import { Body, Controller, Inject, Post, UseGuards } from '@nestjs/common';
+import { ApiCreatedResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { IJwtPayload } from '@novu/shared';
+
+import { CreateLayoutRequestDto, CreateLayoutResponseDto } from './dtos';
+import { CreateLayoutCommand, CreateLayoutUseCase } from './use-cases';
 
 import { JwtAuthGuard } from '../auth/framework/auth.guard';
+import { ExternalApiAccessible } from '../auth/framework/external-api.decorator';
+import { UserSession } from '../shared/framework/user.decorator';
 import { AnalyticsService } from '../shared/services/analytics/analytics.service';
 import { ANALYTICS_SERVICE } from '../shared/shared.module';
 
 @Controller('/layouts')
 @ApiTags('Layouts')
 @UseGuards(JwtAuthGuard)
 export class LayoutsController {
-  constructor(@Inject(ANALYTICS_SERVICE) private analyticsService: AnalyticsService) {}
+  constructor(
+    private createLayoutUseCase: CreateLayoutUseCase,
+    @Inject(ANALYTICS_SERVICE) private analyticsService: AnalyticsService
+  ) {}
+
+  @Post('')
+  @ExternalApiAccessible()
+  @ApiCreatedResponse({
+    type: CreateLayoutResponseDto,
+  })
+  @ApiOperation({ summary: 'Layout creation', description: 'Create a layout' })
+  async createLayout(
+    @UserSession() user: IJwtPayload,
+    @Body() body: CreateLayoutRequestDto
+  ): Promise<CreateLayoutResponseDto> {
+    const layout = await this.createLayoutUseCase.execute(
+      CreateLayoutCommand.create({
+        environmentId: user.environmentId,
+        name: body.name,
+        organizationId: user.organizationId,
+        userId: user._id,
+        content: body.content,
+        variables: body.variables,
```

Comment:
About the variables, I think they should be calculated inside the create usecase, like we do in [create template](https://github.com/novuhq/novu/blob/next/apps/api/src/app/notification-template/usecases/create-notification-template/create-notification-template.usecase.ts#L27)
But we can go over it or I can do it if you want

Reply:
I followed the create message use case logic. I see.
So what should be sent in the body of the creation of the layout? In my head I thought you send the content in one variable with the placeholders for the variables and in another property the variables and their values.

Reply:
As agreed privately, let's address the use case of parsing variables from a string when a user consumes the endpoint in a different PR: https://linear.app/novu/issue/NV-1470/use-case-parse-variables-from-user-content